### PR TITLE
Added a TimeOut Constructor Parameter

### PR DESCRIPTION
**Pull request by PandaBoy00**
Added a TimeOut Parameter with a default of 10 seconds from a TimeSpan Object if the parameter was not supplied during the constructor.

### CODE REVIEW

**Comment by thinkingserious** [comment]
Hi @PandaBoy00,

This one finally bubbled to the top of my queue. Could you please fix the conflicts so I can merge it? Thanks!

**Comment by PandaBoy00** [comment]
Hi Elmer,

Sure no problem.  But what was the conflict?  I was able to compile fine on my end.

Kengie

**Comment by thinkingserious** [comment]
Hello @PandaBoy00,

After I do:

``` bash
git checkout -b PandaBoy00-http-timeout-param master
git pull https://github.com/PandaBoy00/csharp-http-client.git http-timeout-param
```

I get a conflict in `Client.cs`, specifically:

``` csharp
<<<<<<< HEAD
        public WebProxy WebProxy;
=======
        public TimeSpan Timeout;
>>>>>>> a01c011e11a9159b01b7e9147adb4f612defe718
```

**Comment by PandaBoy00** [comment]
Hi Elmer,

I have just resolved the conflict.  Can you please try again?  Thanks.

[cid:643c4b2b-b47b-4c8a-a7fd-4e5c54276d98]

Kengie

**Comment by thinkingserious** [comment]
Hi @PandaBoy00,

It does not look like you have pushed your changes yet. Could you please try it again? Thanks!

**Comment by PandaBoy00** [comment]
Hi Elmer,

Sorry about this.  I push it to the branch http-timeout-param instead of the master branch on my fork.

It is now pushed to the master branch of my fork.  Sorry but I am not too familiar with command line git.  It should be all set now.

Regards,

Kengie

**Comment by thinkingserious** [comment]
Thanks for making the update.

Unfortunately, we have a new set of conflicts:

``` bash
CONFLICT (add/add): Merge conflict in README.md
Auto-merging LICENSE.txt
CONFLICT (add/add): Merge conflict in LICENSE.txt
Auto-merging CONTRIBUTING.md
CONFLICT (add/add): Merge conflict in CONTRIBUTING.md
Auto-merging CHANGELOG.md
CONFLICT (add/add): Merge conflict in CHANGELOG.md
Auto-merging .travis.yml
CONFLICT (add/add): Merge conflict in .travis.yml
Auto-merging .gitignore
CONFLICT (add/add): Merge conflict in .gitignore
Auto-merging .github/ISSUE_TEMPLATE
CONFLICT (add/add): Merge conflict in .github/ISSUE_TEMPLATE
```

Could you please take a look? Thanks!

**Comment by PandaBoy00** [comment]
I have just rechecked in the conflicted files.  Please try again.  Screen Capture for your reference.

**Comment by thinkingserious** [comment]
@PandaBoy00,

Can you please push your changes? Thanks!

**Comment by PandaBoy00** [comment]
Hi there,

Sorry for the troubles.

I have already push my changes to the repo https://github.com/PandaBoy00/csharp-http-client.git.

and everything is already updated.

Please see screen shot for reference.  Is there anything else I'm not doing correctly?  Thanks.

[https://avatars1.githubusercontent.com/u/20187318?v=3&s=400]https://github.com/PandaBoy00/csharp-http-client.git

PandaBoy00/csharp-http-clienthttps://github.com/PandaBoy00/csharp-http-client.git
github.com
csharp-http-client - HTTP client for RESTful APIs, simplified for C#

File: CSharpHTTPClient/Client.cs [hunk 1]
```diff
@@ -90,6 +92,7 @@ public Client(string host, Dictionary<string,string> requestHeaders = null, stri
             }
             Version = (version != null) ? version : null;
             UrlPath = (urlPath != null) ? urlPath : null;
+            Timeout = (timeOut != null) ? (TimeSpan)timeOut : TimeSpan.FromSeconds(10);
```
Comment (thinkingserious) [inline CSharpHTTPClient/Client.cs:104]: Can we make the value passed to TimeSpan.FromSeconds configurable?

**Review by thinkingserious — CHANGES_REQUESTED** [review]
Please see my comment.

With regards to the merge, I'm not sure what is going on. Maybe it's easier to just create a new fork and just replace this file.

**Comment by PandaBoy00** [comment]
Not really sure what you mean from configurable?  Do you mean to get the value from AppSetting?

Currently it is getting a values from the parameter is set, otherwise it will default to 10 seconds.

**Comment by thinkingserious** [comment]
@PandaBoy00,

I'm suggesting that default value be configurable.

**Comment by PandaBoy00** [comment]
Hi there,

I have made the modification based on your suggested and also have check it in.

**Review by thinkingserious — APPROVED** [review]

**Comment by thinkingserious** [comment]
@PandaBoy00,

I see the changes on your branch, but they have not been pushed here. Can you please merge them back here? Thanks!

**Comment by PandaBoy00** [comment]
I have merge both of my branches master and http-timeout-param.  They are both up to date.

Thanks.

File: .travis.yml [hunk 2]
```diff
@@ -18,3 +18,5 @@ notifications:
       Build %{build_number}</a> on branch <i>%{branch}</i> by %{author}: <strong>%{message}</strong>
       <a href="https://github.com/%{repository}/commits/%{commit}">View on GitHub</a>'
     format: html
+
```
Comment (thinkingserious) [inline .travis.yml]: It looks like this edit is causing the tests to fail. Could you please revert this file?

**Review by thinkingserious — CHANGES_REQUESTED** [review]

**Comment by PandaBoy00** [comment]
I have revert the file back to the original from your fork.  You can try again now.

**Review by thinkingserious — APPROVED** [review]

File: CSharpHTTPClient/Client.cs [hunk 3]
```diff
@@ -68,12 +68,17 @@ public class Client : DynamicObject
         public string Version;
         public string UrlPath;
         public string MediaType;
-        public WebProxy WebProxy;
+<<<<<<< HEAD
```
Comment (thinkingserious) [inline CSharpHTTPClient/Client.cs:71]: These lines are the issue.

File: CSharpHTTPClient/Client.cs [hunk 4]
```diff
@@ -106,6 +98,11 @@ public Client(string host, Dictionary<string,string> requestHeaders = null, stri
             }
             Version = (version != null) ? version : null;
             UrlPath = (urlPath != null) ? urlPath : null;
+<<<<<<< HEAD
```
Comment (thinkingserious) [inline CSharpHTTPClient/Client.cs:101]: Here is another case.

**Review by thinkingserious — CHANGES_REQUESTED** [review]

**Review by thinkingserious — APPROVED** [review]

**Comment by thinkingserious** [comment]
@PandaBoy00,

It might be easier to just start over with a new PR.

**Comment by PandaBoy00** [comment]
Sure, I'll do another PR later.  A bit busy recently.

**Comment by thinkingserious** [comment]
Thanks @PandaBoy00!

**Comment by thinkingserious** [comment]
Hi @PandaBoy00,

Just checking in with you :)

**Comment by PandaBoy00** [comment]
Hi Elmer,

Sorry just kinda busy @ work.  I will do another PR asap.

**Comment by thinkingserious** [comment]
Hi @PandaBoy00,

Were you still considering making a new PR?

**Comment by PandaBoy00** [comment]
Hi Elmer,


Yes but I was tied up with work and vacation.  I will do one next week.


________________________________

**Comment by thinkingserious** [comment]
@PandaBoy00,

No worries, thanks for the update and congrats on the vacation!

**Comment by thinkingserious** [comment]
Hi @PandaBoy00,

Just checking in, I hope all is well.

**Comment by PandaBoy00** [comment]
Hi Elmer,


Yeah, I will do another PR today.


________________________________

**Comment by PandaBoy00** [comment]
Moved over to new PR

**Pull request closed without merging**